service/src/main/java/greencity/service/HabitServiceImpl.java

```diff
@@ -163,23 +161,40 @@ public void update(HabitManagementDto habitManagementDto, MultipartFile image) {
         Habit habit = habitRepo.findById(habitManagementDto.getId())
             .orElseThrow(() -> new WrongIdException(ErrorMessage.HABIT_NOT_FOUND_BY_ID));
 
-        habit.getHabitTranslations()
-            .forEach(ht -> {
-                HabitTranslationManagementDto htmd = habitManagementDto.getHabitTranslations().stream()
-                    .filter(e -> e.getLanguageCode().equals(ht.getLanguage().getCode())).findFirst()
-                    .orElseThrow(RuntimeException::new);
-
-                ht.setDescription(htmd.getDescription());
-                ht.setHabitItem(htmd.getHabitItem());
-                ht.setName(htmd.getName());
-            });
-        habit.setImage(habitManagementDto.getImage());
+        Map<String, HabitTranslationManagementDto> translationDtoMap = getMapTranslationsDtos(habitManagementDto);
+        habit.getHabitTranslations().forEach(
+            ht -> updateHabitTranslation(translationDtoMap.get(ht.getLanguage().getCode()), ht));
 
         uploadImageForHabit(habitManagementDto, image, habit);
 
         habitRepo.save(habit);
     }
 
+    /**
+     * Method updates {@link HabitTranslation} with {@link HabitTranslationManagementDto} fields.
+     *
+     * @param htDto {@link HabitTranslationManagementDto} instance.
+     * @param ht    {@link HabitTranslation} instance.
+     */
+    private void updateHabitTranslation(HabitTranslationManagementDto htDto, HabitTranslation ht) {
```

Comment:
bad method naming, update is used when smth is updatedin DB, for example. Please, rename it to smth like enhanceWith...

Reply:
enhanceTranslationWithDto - would be great?